.github/workflows/pipeline_main.yml

```diff
@@ -0,0 +1,97 @@
+name: Main pipeline [build -> test -> deploy]
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  packages: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: false
+
+jobs:
+
+  validate:
+    uses: ./.github/workflows/validate.yml
+
+  helm_lint:
+    name: Helm lint
+    strategy:
+      matrix:
+        environments: [ 'dev', 'test', 'preprod', 'prod' ]
+    uses: ministryofjustice/hmpps-github-actions/.github/workflows/test_helm_lint.yml@v2
+    secrets: inherit
+    with:
+      environment: ${{ matrix.environments }}
+
+  build_docker:
+    name: Build docker image
+    uses: ministryofjustice/hmpps-github-actions/.github/workflows/docker_build.yml@v2
+    with:
+      docker_registry: 'ghcr.io'
+      registry_org: 'ministryofjustice'
+      tag_latest: false
+      push: false
+      load: true
+      docker_multiplatform: false
+      upload_image_artifact: true
```

Comment:
do we need to specify image artifact retention here? Or is there some default we're happy with already?

Reply:
The default for this is to set it to false, we want to retain it AFAIK

`      upload_image_artifact:
        description: Upload image to artifacts in GitHub Actions
        required: false
        type: boolean
        default: false`

Reply:
sorry, I meant do we need to add a new line here specifying artifact retention duration - there is one in https://github.com/ministryofjustice/hmpps-assess-risks-and-needs-handover-service/blob/main/.github/workflows/pipeline_main.yml